src/internal/clusterstate/migrationutils/pg_batcher.go

```diff
@@ -183,3 +183,47 @@ func (pb *postgresBatcher) execute(ctx context.Context) error {
 func (pb *postgresBatcher) Finish(ctx context.Context) error {
 	return pb.execute(ctx)
 }
+
+type SimplePostgresBatcher struct {
```

Comment:
This seems like a reasonable implementation.  If it's going to be public and you want it to be reused, it should have comments.  e.g. The caller needs to handle all errors from Add and Close.  You might rename Close to Flush since the Batcher continues to work after calling Close.